src/platform_impl/macos/util/async.rs

```diff
@@ -218,11 +222,17 @@ pub(crate) fn set_title_async(window: &NSWindow, title: String) {
 //
 // ArturKovacs: It's important that this operation keeps the underlying window alive
 // through the `Id` because otherwise it would dereference free'd memory
-pub(crate) fn close_async(window: Id<NSWindow, Shared>) {
+pub(crate) fn close_async(window: Id<WinitWindow, Shared>) {
     let window = MainThreadSafe(window);
     Queue::main().exec_async(move || {
         autoreleasepool(move |_| {
             window.close();
+
+            let event = Event::WindowEvent {
+                window_id: WindowId(window.id()),
+                event: WindowEvent::Destroyed,
+            };
+            AppState::queue_event(EventWrapper::StaticEvent(event));
```

Comment:
Nit: Use `window.emit_event`

Reply:
Only `WinitWindowDelegate` has this method. Should I add such method to `WinitWindow` too? Or could I just access `WinitWindowDelegate` by `pub(crate)` it.

Reply:
Just do `pub(crate)`

Reply:
@madsmtm I got a real weird behavior if I move a shared pointer of `delegate` into `exec_async`, `windowWillClose` will be called somehow!
So I can just call `delegate.window.close()` instead in latest commit. Wdyt?

Reply:
I found the delegate has to be used in `exec_async` somehow, so `windowWillCose` can be invoked.
So I call `delegate.window.close()` instead which should look a little bit cleaner.